ydb/core/kqp/host/kqp_gateway_proxy.cpp

```diff
@@ -792,7 +793,86 @@ class TKqpGatewayProxy : public IKikimrGateway {
     TFuture<TGenericResult> ModifyPermissions(const TString& cluster,
         const TModifyPermissionsSettings& settings) override
     {
-        FORWARD_ENSURE_NO_PREPARE(ModifyPermissions, cluster, settings);
+        CHECK_PREPARED_DDL(ModifyPermissions);
+
+        auto modifyPermissionsPromise = NewPromise<TGenericResult>();
+
+        if (settings.Permissions.empty() && !settings.IsPermissionsClear) {
+            return MakeFuture(ResultFromError<TGenericResult>("No permissions names for modify permissions"));
+        }
+
+        if (settings.Pathes.empty()) {
+            return MakeFuture(ResultFromError<TGenericResult>("No pathes for modify permissions"));
+        }
+
+        if (settings.Roles.empty()) {
+            return MakeFuture(ResultFromError<TGenericResult>("No roles for modify permissions"));
+        }
+
+        NACLib::TDiffACL acl;
+        switch (settings.Action) {
+            case NYql::TModifyPermissionsSettings::EAction::Grant: {
+                for (const auto& sid : settings.Roles) {
+                    for (const auto& permission : settings.Permissions) {
+                        TACLAttrs aclAttrs = ConvertYdbPermissionNameToACLAttrs(permission);
+                        acl.AddAccess(NACLib::EAccessType::Allow, aclAttrs.AccessMask, sid, aclAttrs.InheritanceType);
+                    }
+                }
+            }
+            break;
```

Comment:
Here and futher in this file. I think it's more clear to write `break;` within `{ ... }` for current case statement